lib/server/monalisaSynchronizers/DataPassesSynchronizer.js

```diff
@@ -0,0 +1,181 @@
+            'SELECT name, max(dpr.run_number) as last_run_in_details \
+            FROM data_passes AS dp \
+            LEFT JOIN data_passes_runs AS dpr \
+                ON dpr.data_pass_id = dp.id \
+            GROUP BY name;');
+
+        const lastRunsEntries = lastRunsData.map((lastRunData) => {
+            const { name, last_run_in_details: lastRun } = lastRunData;
+            return [name, lastRun];
+        });
+
+        return {
+            lastRuns: Object.fromEntries(lastRunsEntries),
+            yearLowerLimit: this.yearLowerLimit,
+        };
+    }
+
+    /**
+     * Check wether data of one data pass are valid
+     * @param {object} dataPass data pass data
+     * @param {object} context context of synchronization
+     * @return {boolean} true if data pass is valid
+     */
+    needToUpdateDataPass(dataPass, context) {
+        return dataPass.lhcPeriod?.year >= context.yearLowerLimit &&
+            (!dataPass.lastRun ||
+            dataPass.lastRun !== context.lastRuns[dataPass.name]);
+    }
+
+    /**
+     * Adjust data pass data to desired format
+     * @param {object} dataPass data pass
+     * @return {object} adjusted data pass
+     */
+    adjustSingleDataPass(dataPass) {
+        const { lhcPeriod } = dataPass;
+        dataPass = mapObjectProperties(dataPass, this.fieldsMapping);
+        dataPass.outputSize = dataPass.outputSize ? Number(dataPass.outputSize) : null;
+        dataPass.lhcPeriod = lhcPeriod;
+        return dataPass;
+    }
+
+    /**
+     * Used to process raw response to custom format, remove unnecessary data
+     * @param {object} rawResponse raw data acquired from external service
+     * @param {object} context context of synchronization
+     * @return {object[]} adjusted data
+     */
+    async processRawResponse(rawResponse, context) {
+        return Object.entries(rawResponse)
+            .map(([dataPassName, dataPass]) => {
+                dataPass['name'] = dataPassName.trim();
+                return dataPass;
+            })
+            .filter((dataPass) => dataPass.name.match(RegExp(`${LHC_PERIOD_NAME_REGEX.source}_`)))
+            .map((dataPass) => {
+                dataPass.lhcPeriod = extractLhcPeriod(dataPass.name);
+                return dataPass;
+            })
+            .filter((dataPass) => this.needToUpdateDataPass(dataPass, context))
+            .map((dataPass) => this.adjustSingleDataPass(dataPass));
+    }
+
+    /**
+     * Implements logic for inserting/updating database
+     * @param {object[]} dataPasses data passes list
+     * @return {Promise<void>} promise
+     */
+    async executeDbAction(dataPasses) {
+        this.dataPasses = dataPasses;
+        // eslint-disable-next-line require-jsdoc
+        const pipeline = (dataPass) => LhcPeriodRepository
+            .findOne({ where: { name: dataPass.lhcPeriod.name } })
+            .then((lhcPeriod) => DataPassRepository.upsert({ lhcPeriodId: lhcPeriod.id, ...dataPass }))
```

Comment:
Use await, it's more clear (and it's the common way we do in bkp)

Reply:
applied